src/main/java/com/coveo/pushapiclient/BaseSource.java

```diff
@@ -0,0 +1,18 @@
+package com.coveo.pushapiclient;
+
+public interface BaseSource {
+    /**
+     * Return an instance of {@link PlatformClient}
+     *
+     * @return
+     */
+    PlatformClient getPlatformClient();
```

Comment:
It feels strange for the Source class to expose a platform client. Should it be kept internal, and have the class expose only relevant methods?

Reply:
I completely agree that it feels strange and would be open to change it. I kept the PlatformClient getter because:
1. It was already exposed on the (to be) deprecated `Source` class. 
2. The upload services (`StreamService`, `PushService`) need an instance of PlatformClient to perform requests against the API (open stream, create S3 buckets, create stream chunk, upload content, ...).
```java
StreamService service = new StreamService(source)
```

To avoid exposing the platform client, the source classes would need to have getters for the API key and source url. 
```java
String key = source.getApiKey();
String url = source.getSourceUrl();
```
Do you see any issues with a getter returning the API key?

Reply:
Honestly, I think it looks nice. It makes it clear what values we expect to get from the source class, and aligns with what is shown in the AdminUI.